### PR TITLE
Affixing exit summary to publish-docs command

### PR DESCRIPTION
**Pull request by amoghrajesh**
It would be nice to have a summary of the action all provider docs underwent at the end of `breeze publish-docs` command as it is super easy to miss the notes while publishing in bulk

Sample output:
```
(new-env) ➜  airflow git:(summaryPublishDocs) ✗ breeze release-management publish-docs apache.pinot apache.beam apache.impala --include-success-outputs
Publishing docs for 3 package(s)
 - apache-airflow-providers-apache-beam
 - apache-airflow-providers-apache-impala
 - apache-airflow-providers-apache-pinot

Publishing docs for apache-airflow-providers-apache-beam
Copy directory: /Users/adesai/Documents/OSS/airflow/docs/_build/docs/apache-airflow-providers-apache-beam/stable => 
/Users/adesai/Documents/OSS/airflow-site/docs-archive/apache-airflow-providers-apache-beam/5.6.0
Skipping previously existing /Users/adesai/Documents/OSS/airflow-site/docs-archive/apache-airflow-providers-apache-beam/5.6.0! Delete it manually if you want to regenerate it!

Publishing docs for apache-airflow-providers-apache-impala
Copy directory: /Users/adesai/Documents/OSS/airflow/docs/_build/docs/apache-airflow-providers-apache-impala/stable => 
/Users/adesai/Documents/OSS/airflow-site/docs-archive/apache-airflow-providers-apache-impala/1.3.0
Skipping previously existing /Users/adesai/Documents/OSS/airflow-site/docs-archive/apache-airflow-providers-apache-impala/1.3.0! Delete it manually if you want to regenerate it!

Publishing docs for apache-airflow-providers-apache-pinot
Copy directory: /Users/adesai/Documents/OSS/airflow/docs/_build/docs/apache-airflow-providers-apache-pinot/stable => 
/Users/adesai/Documents/OSS/airflow-site/docs-archive/apache-airflow-providers-apache-pinot/4.3.0
Skipping previously existing /Users/adesai/Documents/OSS/airflow-site/docs-archive/apache-airflow-providers-apache-pinot/4.3.0! Delete it manually if you want to regenerate it!

Success entries:
Skipped entries:
Skipping apache-airflow-providers-apache-beam: Previously existing directory
Skipping apache-airflow-providers-apache-impala: Previously existing directory
Skipping apache-airflow-providers-apache-pinot: Previously existing directory
(new-env) ➜  airflow git:(summaryPublishDocs) ✗ breeze release-management publish-docs apache.pinot apache.beam apache.impala apache.spark  --include-success-outputs
(new-env) ➜  airflow git:(summaryPublishDocs) ✗ breeze release-management publish-docs apache.pinot apache.beam apache.impala apache.spark  --include-success-outputs
Publishing docs for 4 package(s)
 - apache-airflow-providers-apache-beam
 - apache-airflow-providers-apache-impala
 - apache-airflow-providers-apache-pinot
 - apache-airflow-providers-apache-spark

Publishing docs for apache-airflow-providers-apache-beam
Copy directory: /Users/adesai/Documents/OSS/airflow/docs/_build/docs/apache-airflow-providers-apache-beam/stable => 
/Users/adesai/Documents/OSS/airflow-site/docs-archive/apache-airflow-providers-apache-beam/5.6.0
Skipping previously existing /Users/adesai/Documents/OSS/airflow-site/docs-archive/apache-airflow-providers-apache-beam/5.6.0! Delete it manually if you want to regenerate it!

Publishing docs for apache-airflow-providers-apache-impala
Copy directory: /Users/adesai/Documents/OSS/airflow/docs/_build/docs/apache-airflow-providers-apache-impala/stable => 
/Users/adesai/Documents/OSS/airflow-site/docs-archive/apache-airflow-providers-apache-impala/1.3.0
Skipping previously existing /Users/adesai/Documents/OSS/airflow-site/docs-archive/apache-airflow-providers-apache-impala/1.3.0! Delete it manually if you want to regenerate it!

Publishing docs for apache-airflow-providers-apache-pinot
Copy directory: /Users/adesai/Documents/OSS/airflow/docs/_build/docs/apache-airflow-providers-apache-pinot/stable => 
/Users/adesai/Documents/OSS/airflow-site/docs-archive/apache-airflow-providers-apache-pinot/4.3.0
Skipping previously existing /Users/adesai/Documents/OSS/airflow-site/docs-archive/apache-airflow-providers-apache-pinot/4.3.0! Delete it manually if you want to regenerate it!

Publishing docs for apache-airflow-providers-apache-spark
Copy directory: /Users/adesai/Documents/OSS/airflow/docs/_build/docs/apache-airflow-providers-apache-spark/stable => 
/Users/adesai/Documents/OSS/airflow-site/docs-archive/apache-airflow-providers-apache-spark/4.7.1

Success entries:
Docs published: apache-airflow-providers-apache-spark
=================================================

Skipped entries:
Skipping apache-airflow-providers-apache-beam: Previously existing directory
Skipping apache-airflow-providers-apache-impala: Previously existing directory
Skipping apache-airflow-providers-apache-pinot: Previously existing directory
```

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by amoghrajesh** [comment]
Ask by @eladkal for ease of doc publishing

**Review by dirrao — APPROVED** [review]

**Review by potiuk — APPROVED** [review]
LGTM. @eladkal ?

**Review by potiuk — CHANGES_REQUESTED** [review]
Actually one thing should be changed. Rather than ysing print() you should use `get_console().print()` (and colors - `[success]` for published and `[warning]` for skipped + plus use built-in `get_console().rule()` to add separator instead of `===`

**Comment by amoghrajesh** [comment]
> Actually one thing should be changed. Rather than ysing print() you should use `get_console().print()` (and colors - `[success]` for published and `[warning]` for skipped + plus use built-in `get_console().rule()` to add separator instead of `===`

Sure let me handle that

**Comment by amoghrajesh** [comment]
Good feedback @potiuk

this is how it looks now: (I am quite impressed actually)
```
(new-env) ➜  airflow git:(summaryPublishDocs) ✗ breeze release-management publish-docs apache.pinot apache.beam apache.impala apache.spark apache.pig microsoft.winrm --include-success-outputs
Publishing docs for 6 package(s)
 - apache-airflow-providers-apache-beam
 - apache-airflow-providers-apache-impala
 - apache-airflow-providers-apache-pig
 - apache-airflow-providers-apache-pinot
 - apache-airflow-providers-apache-spark
 - apache-airflow-providers-microsoft-winrm

Publishing docs for apache-airflow-providers-apache-beam
Copy directory: /Users/adesai/Documents/OSS/airflow/docs/_build/docs/apache-airflow-providers-apache-beam/stable => 
/Users/adesai/Documents/OSS/airflow-site/docs-archive/apache-airflow-providers-apache-beam/5.6.0
Skipping previously existing /Users/adesai/Documents/OSS/airflow-site/docs-archive/apache-airflow-providers-apache-beam/5.6.0! Delete it manually if you want to regenerate it!

Publishing docs for apache-airflow-providers-apache-impala
Copy directory: /Users/adesai/Documents/OSS/airflow/docs/_build/docs/apache-airflow-providers-apache-impala/stable => 
/Users/adesai/Documents/OSS/airflow-site/docs-archive/apache-airflow-providers-apache-impala/1.3.0
Skipping previously existing /Users/adesai/Documents/OSS/airflow-site/docs-archive/apache-airflow-providers-apache-impala/1.3.0! Delete it manually if you want to regenerate it!

Publishing docs for apache-airflow-providers-apache-pig
Copy directory: /Users/adesai/Documents/OSS/airflow/docs/_build/docs/apache-airflow-providers-apache-pig/stable => 
/Users/adesai/Documents/OSS/airflow-site/docs-archive/apache-airflow-providers-apache-pig/4.3.0
Skipping previously existing /Users/adesai/Documents/OSS/airflow-site/docs-archive/apache-airflow-providers-apache-pig/4.3.0! Delete it manually if you want to regenerate it!

Publishing docs for apache-airflow-providers-apache-pinot
Copy directory: /Users/adesai/Documents/OSS/airflow/docs/_build/docs/apache-airflow-providers-apache-pinot/stable => 
/Users/adesai/Documents/OSS/airflow-site/docs-archive/apache-airflow-providers-apache-pinot/4.3.0
Skipping previously existing /Users/adesai/Documents/OSS/airflow-site/docs-archive/apache-airflow-providers-apache-pinot/4.3.0! Delete it manually if you want to regenerate it!

Publishing docs for apache-airflow-providers-apache-spark
Copy directory: /Users/adesai/Documents/OSS/airflow/docs/_build/docs/apache-airflow-providers-apache-spark/stable => 
/Users/adesai/Documents/OSS/airflow-site/docs-archive/apache-airflow-providers-apache-spark/4.7.1
Skipping previously existing /Users/adesai/Documents/OSS/airflow-site/docs-archive/apache-airflow-providers-apache-spark/4.7.1! Delete it manually if you want to regenerate it!

Publishing docs for apache-airflow-providers-microsoft-winrm
Copy directory: /Users/adesai/Documents/OSS/airflow/docs/_build/docs/apache-airflow-providers-microsoft-winrm/stable => 
/Users/adesai/Documents/OSS/airflow-site/docs-archive/apache-airflow-providers-microsoft-winrm/3.4.0
Skipping previously existing /Users/adesai/Documents/OSS/airflow-site/docs-archive/apache-airflow-providers-microsoft-winrm/3.4.0! Delete it manually if you want to regenerate it!

Packages those were published:
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────

Packages those were skipped:
Skipping apache-airflow-providers-apache-beam: Previously existing directory
Skipping apache-airflow-providers-apache-impala: Previously existing directory
Skipping apache-airflow-providers-apache-pig: Previously existing directory
Skipping apache-airflow-providers-apache-pinot: Previously existing directory
Skipping apache-airflow-providers-apache-spark: Previously existing directory
Skipping apache-airflow-providers-microsoft-winrm: Previously existing directory
```

**Comment by amoghrajesh** [comment]
Woops my bad, it has coloring. This is a screenshot
image

File: dev/breeze/src/airflow_breeze/commands/release_management_commands.py [hunk 1]
```diff
@@ -1230,14 +1229,23 @@ def run_publish_docs_in_parallel(
...
+            # Iterate over the results and collect success and skipped entries
+            for index, result in enumerate(results):
+                return_code, message = result.get()
+                if return_code == 0:
+                    success_entries.append(message)
+                else:
+                    skipped_entries.append(message)
+
+    if include_success_outputs:
+        get_console().print("[success]Packages those were published:")
```
Comment (potiuk) [inline dev/breeze/src/airflow_breeze/commands/release_management_commands.py]:
```suggestion
        get_console().print("[success]Packages published:")
```

File: dev/breeze/src/airflow_breeze/commands/release_management_commands.py [hunk 2]
```diff
@@ -1230,14 +1229,23 @@ def run_publish_docs_in_parallel(
...
+        for entry in success_entries:
+            get_console().print(f"[success]{entry}")
+        get_console().rule()
+    get_console().print("\n[warning]Packages those were skippeds:")
```
Comment (potiuk) [inline dev/breeze/src/airflow_breeze/commands/release_management_commands.py]:
```suggestion
    get_console().print("\n[warning]Packages skippeds:")
```

File: dev/breeze/src/airflow_breeze/commands/release_management_commands.py [hunk 3]
```diff
@@ -1307,10 +1315,24 @@ def publish_docs(
...
-            run_docs_publishing(
+            return_code, message = run_docs_publishing(
                 package_name, airflow_site_directory, override_versioned, verbose=get_verbose(), output=None
             )
+            if return_code == 0:
+                success_entries.append(message)
+            else:
+                skipped_entries.append(message)
+        if include_success_outputs:
+            get_console().print("[success]Packages those were published:")
```
Comment (potiuk) [inline dev/breeze/src/airflow_breeze/commands/release_management_commands.py]:
```suggestion
            get_console().print("[success]Packages published:")
```

File: dev/breeze/src/airflow_breeze/commands/release_management_commands.py [hunk 4]
```diff
@@ -1307,10 +1315,24 @@ def publish_docs(
...
+            for entry in success_entries:
+                get_console().print(f"[success]{entry}")
+            get_console().rule()
+        get_console().print("\n[warning]Packages those were skipped:")
```
Comment (potiuk) [inline dev/breeze/src/airflow_breeze/commands/release_management_commands.py]:
```suggestion
        get_console().print("\n[warning]Packages skipped:")
```

**Review by potiuk — APPROVED** [review]
Nits only.

**Pull request merged**